internal/providers/terraform/azure/function_app.go

```diff
@@ -0,0 +1,188 @@
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetAzureRMAppFunctionRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_function_app",
+		RFunc: NewAzureRMAppFunction,
+		ReferenceAttributes: []string{
+			"app_service_plan_id",
+		},
+	}
+}
+
+func NewAzureRMAppFunction(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	meterNameCPU := "vCPU Duration"
+	meterNameMemory := "Memory Duration"
+	meterNameExecTime := "Execution Time"
+	meterNameExecutions := "Total Executions"
+	memorySize := 1.0
```

Comment:
These never change in this file so you can just hard-code them in the functions that use them instead of defining variables and passing them around. e.g. on line 115, the string can be set directly without using any variables. That makes the code easier to read.